src/main/java/seedu/address/model/person/UniquePersonList.java

```diff
@@ -97,6 +97,40 @@ public void setPersons(List<Person> persons) {
         internalList.setAll(persons);
     }
 
+    /**
+     * Return person with the given nric.
+     * @param nric
+     * @return
+     */
+    public Person get(String nric) {
+        for (Person p : internalList) {
+            if (p.getNric().value.equals(nric)) {
+                return p;
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Returns true if the list contains a link between the given patient and caregiver.
+     * @param patient
+     * @param caregiver
+     * @return
+     */
+    public boolean hasLink(Person patient, Person caregiver) {
+        return patient.hasCaregiver(caregiver);
+    }
+
+    /**
+     * Adds a link between the given patient and caregiver.
+     * @param patient
+     * @param caregiver
+     */
+    public void addLink(Person patient, Person caregiver) {
+        patient.addCaregiver(caregiver);
+        caregiver.addPatient(patient);
+    }
```

Comment:
A few points
1. Perhaps indicate the successful operation to the caller of this method by returning a boolean
2. Handle the case of either of the inputs being null
3. Also handle the case if either of the persons is not in the list (possibly by throwing an exception?)